walk/walk_test.go

```diff
@@ -684,6 +684,57 @@ func TestSubdirsContained(t *testing.T) {
 	})
 }
 
+func TestRelsToVisit(t *testing.T) {
+	dir, cleanup := testtools.CreateFiles(t, []testtools.FileSpec{
+		{Path: "update/sub/"},
+		{Path: "extra/a/b/"},
+	})
+	defer cleanup()
+
+	// Update the update/ directory only without recursing.
+	// Return extra/a/ through RelsToVisit.
+	var configuredRels, visitedRels, updatedRels []string
+	c, cexts := testConfig(t, dir)
+	cexts = append(cexts, &testConfigurer{
+		configure: func(_ *config.Config, rel string, _ *rule.File) {
+			configuredRels = append(configuredRels, rel)
+		},
+	})
+	updateDir := filepath.Join(dir, "update")
+	err := Walk2(c, cexts, []string{updateDir}, UpdateDirsMode, func(args Walk2FuncArgs) Walk2FuncResult {
+		visitedRels = append(visitedRels, args.Rel)
+		if args.Update {
+			updatedRels = append(updatedRels, args.Rel)
+		}
+		res := Walk2FuncResult{}
+		if args.Rel == "update" {
```

Comment:
Could you extend this test to also return an additional dir to visit when `extra/a` is visited? Perhaps also include `update` to verify that we don't get into an endless loop.

Reply:
Yes, done. Also verified that parents of non-existent directories are configured (now fixed).